rootfs/etc/nginx/lua/test/certificate_test.lua

```diff
@@ -0,0 +1,50 @@
+_G._TEST = true
+
+describe("Certificate", function()
+    local certificate = require("certificate")
+    local ssl = require("ngx.ssl")
+    local match = require("luassert.match")
+    
+    describe("set_pem_cert_key", function()
+        ssl.cert_pem_to_der = function(cert) return cert end
+        ssl.set_der_cert = function(cert) return cert end
+        ssl.priv_key_pem_to_der = function(priv_key) return priv_key end
+        ssl.set_der_priv_key = function(priv_key) return priv_key end
+
+        local mock_cert_key = "fake_pem_cert_key"
+
+        it("successfully sets the PEM encoded certificate and private key", function()
+            spy.on(ssl, "set_der_cert")
+            spy.on(ssl, "set_der_priv_key")
+            
+            local set_cert_key_err = certificate.set_pem_cert_key(mock_cert_key)
+            assert.spy(ssl.set_der_cert).was_called_with(mock_cert_key)
+            assert.spy(ssl.set_der_priv_key).was_called_with(mock_cert_key)
+            assert.is_not_truthy(set_cert_key_err)
+        end)
+    end)
+
+    describe("call", function()
+        ssl.server_name = function() return "hostname" end
+        ssl.clear_certs = function() return true, "" end
+
+        it("does not clear fallback certificates and logs error message when pem_cert_key does not exist", function()
+            spy.on(ngx, "log")
+            spy.on(ssl, "clear_certs")
+
+            assert.has_no.errors(certificate.call)
+            assert.spy(ngx.log).was_called_with(ngx.ERR, "Certificate not found for the given hostname: hostname")
+            assert.spy(ssl.clear_certs).was_not_called()
+        end)
+
+        it("successfully sets SSL certificate and key when hostname is found in dictionary", function()
+            local _ = match._
+            local fake_pem_cert_key = "fake_pem_cert_key"
+            ngx.shared.certificate_data:set("hostname", fake_pem_cert_key)
+
+            spy.on(ngx, "log")
+            assert.has_no.errors(certificate.call)
+            assert.spy(ngx.log).was_not_called_with(ngx.ERR, _)
```

Comment:
I'm trying to add an assertion that `set_pem_cert_key` is being called with `fake_pem_cert_key`, but `set_pem_cert_key` never gets called for some reason:
```
spy.on(certificate, "set_pem_cert_key")
assert.spy(certificate.set_pem_cert_key).was_not_called_with(fake_pem_cert_key)
```